File: backend/ShopApi/Dtos/AuthDtos.cs
using System.ComponentModel.DataAnnotations;

namespace ShopApi.Dtos;

public record RegisterRequest(
    [Required, StringLength(100, MinimumLength = 2)] string Username,
    [Required, EmailAddress, StringLength(255)] string Email,
    [Required, StringLength(100, MinimumLength = 6)] string Password);

public record LoginRequest(
    [Required, EmailAddress, StringLength(255)] string Email,
    [Required] string Password);

public record UserResponse(
    int Id,
    string Username,
    string Email,
    string Role);

public record LoginResponse(
    string Token,
    UserResponse User);

File: app/src/main/java/com/example/shop/data/remote/dto/AuthDtos.kt
package com.example.shop.data.remote.dto

data class RegisterRequest(
    val username: String,
    val email: String,
    val password: String
)

data class LoginRequest(
    val email: String,
    val password: String
)

data class GoogleLoginRequest(
    val idToken: String
)

data class ApiUserResponse(
    val id: Int,
    val username: String,
    val email: String,
    val role: String
)

data class LoginResponse(
    val token: String,
    val user: ApiUserResponse
)
